posthog/special_migrations/test/test_utils.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+
+from posthog.constants import AnalyticsDBMS
+from posthog.models.special_migration import MigrationStatus
+from posthog.models.team import Team
+from posthog.special_migrations.setup import SPECIAL_MIGRATIONS_EXAMPLE_MODULE_PATH
+from posthog.special_migrations.test.util import create_special_migration
+from posthog.special_migrations.utils import (
+    complete_migration,
+    execute_op,
+    force_stop_migration,
+    process_error,
+    trigger_migration,
+)
+from posthog.test.base import BaseTest
+
+
+class TestUtils(BaseTest):
+    @pytest.mark.ee
+    @patch("ee.clickhouse.client.sync_execute")
+    def test_execute_op_clickhouse(self, mock_sync_execute):
+        execute_op(AnalyticsDBMS.CLICKHOUSE, "SELECT 1", 10, "some_id")
+
+        # correctly routes to ch
+        mock_sync_execute.assert_called_once_with("/* some_id */ SELECT 1", settings={"max_execution_time": 10})
+
+    @patch("django.db.connection.cursor")
+    def test_execute_op_postgres(self, mock_cursor):
+        execute_op(AnalyticsDBMS.POSTGRES, "SELECT 1", 10, "some_id")
+
+        # correctly routes to postgres
+        mock_cursor.assert_called_once()
+
+    @patch("posthog.special_migrations.runner.attempt_migration_rollback")
+    def test_process_error(self, _):
+        sm = create_special_migration()
+        process_error(sm, "some error")
+
+        sm.refresh_from_db()
+        self.assertEqual(sm.status, MigrationStatus.Errored)
+        self.assertEqual(sm.last_error, "some error")
+        self.assertEqual(sm.finished_at.day, datetime.now().day)
```

Comment:
this can be flaky at the end of the day

Reply:
highly unlikely but yes, I thought about this. will consider a better way although the conditions for this to fail would have to be us crossing to a new day within a some milliseconds after completing the migration

Reply:
you can use some time range check, e.g. `now - timedelta(3h) < finished_at < now + timedelta(3h)`